subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1170,10 +1170,58 @@ BT_GATT_SERVICE_DEFINE(bass_svc,
 	RECEIVE_STATE_CHARACTERISTIC(2)
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 2 */
 #endif /* CONFIG_BT_BAP_SCAN_DELEGATOR_RECV_STATE_COUNT > 1 */
-);
+};
+
+static struct bt_gatt_service bass_svc = BT_GATT_SERVICE(attr_bass_svc);
+
+static bool scan_delegator_registered;
+static bool pa_sync_cb_registered;
```

Comment:
Suggest to use `ATOMIC_DEFINE` for these instead so that we can better be thread safe.

Reply:
Done.